### PR TITLE
feat: update number of lines for finished step

### PR DESCRIPTION
**Pull request by minzcmu**
- update number of total lines when a step finishes.

Tested it out with in-a-box. Works!
screen shot 2018-07-24 at 4 15 46 pm

Related to https://github.com/screwdriver-cd/screwdriver/issues/1114

### CODE REVIEW

File: logservice_test.go [hunk 1]
```diff
@@ -89,12 +101,13 @@ func newAppFromEmitter(emitterPath string) App {
...
-	run         func()
-	logReader   func() io.Reader
-	uploader    func() sdstoreuploader.SDStoreUploader
-	archiveLogs func(uploader sdstoreuploader.SDStoreUploader, src io.Reader) error
-	stepSaver   func(step string) StepSaver
-	buildID     string
+	run         	func()
+	logReader   	func() io.Reader
+	uploader    	func() sdstoreuploader.SDStoreUploader
+	screwdriverapi  func() screwdriver.API
```
Comment (tkyi) [inline logservice_test.go]: should probably be screwdriverAPI

File: screwdriver/screwdriver.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
+package screwdriver
```
Comment (minzcmu) [inline screwdriver/screwdriver.go:1]: Most code of this file is from https://github.com/screwdriver-cd/launcher/blob/master/screwdriver/screwdriver.go

File: stepsaver.go [hunk 3]
```diff
@@ -161,8 +169,8 @@ func (s *stepSaver) Save() error {
 }
 
 // NewStepSaver creates a StepSaver out of a name and sdstoreuploader.SDStoreUploader
-func NewStepSaver(name string, uploader sdstoreuploader.SDStoreUploader, linesPerFile int) StepSaver {
-	s := &stepSaver{StepName: name, Uploader: uploader, ticker: time.NewTicker(uploadInterval), linesPerFile: linesPerFile}
+func NewStepSaver(name string, uploader sdstoreuploader.SDStoreUploader, linesPerFile int, screwdriverapi screwdriver.API) StepSaver {
```
Comment (tkyi) [inline stepsaver.go]: also should probably be `screwdriverAPI screwdriver.API`

File: logservice_test.go [hunk 4]
```diff
@@ -68,6 +69,17 @@ func (m *mockSDStoreUploader) Upload(path string, filePath string) error {
 	return nil
 }
 
+type mockScrewdriverAPI struct {
+	updateSteplines func(string, int) error
```
Comment (tkyi) [inline logservice_test.go]: for consistency, also should have everything be camel case for `updateStepLines`

File: screwdriver/screwdriver.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    buildID string
+	baseURL string
+	token   string
+	client  *http.Client
+}
+
+// New returns a new API object
+func New(buildID, url, token string) (API, error) {
+	newapi := api{
+        buildID,
```
Comment (tkyi) [inline screwdriver/screwdriver.go]: indentation

File: screwdriver/screwdriver.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+	Reason     string `json:"error"`
+	Message    string `json:"message"`
+}
+
+func (e SDError) Error() string {
+	return fmt.Sprintf("%d %s: %s", e.StatusCode, e.Reason, e.Message)
+}
+
+type api struct {
+    buildID string
```
Comment (tkyi) [inline screwdriver/screwdriver.go]: indentation

**Review by d2lam — APPROVED** [review]
Looks ok logic wise 👍

**Review by tkyi — APPROVED** [review]
Small comments 👍

File: screwdriver/screwdriver.go [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+type api struct {
+	buildID string
+	baseURL string
+	token	string
+	client	*http.Client
+}
+
+// New returns a new API object
+func New(buildID, url, token string) (API, error) {
+	newapi := api{
```
Comment (tkyi) [inline screwdriver/screwdriver.go]: newAPI*

File: screwdriver/screwdriver.go [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+		buildID,
+		url,
+		token,
+		&http.Client{Timeout: 10 * time.Second},
+	}
+	return API(newapi), nil
```
Comment (tkyi) [inline screwdriver/screwdriver.go]: newAPI

File: screwdriver/screwdriver.go [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+// StepStartPayload is a Screwdriver Step Start payload.
+type SteplinesPayload struct {
```
Comment (tkyi) [inline screwdriver/screwdriver.go]: StepLines

**Review by tkyi — APPROVED** [review]
👍

**Pull request merged**